src/vec2.js

```diff
@@ -476,33 +476,13 @@ export function rotate(out, a, b, rad) {
  * @returns {Number} The angle in radians
  */
 export function angle(a, b) {
-  let x1 = a[0],
-    y1 = a[1],
-    x2 = b[0],
-    y2 = b[1];
-
-  let len1 = x1*x1 + y1*y1;
-  if (len1 > 0) {
-    //TODO: evaluate use of glm_invsqrt here?
-    len1 = 1 / Math.sqrt(len1);
-  }
-
-  let len2 = x2*x2 + y2*y2;
-  if (len2 > 0) {
-    //TODO: evaluate use of glm_invsqrt here?
-    len2 = 1 / Math.sqrt(len2);
-  }
-
-  let cosine = (x1 * x2 + y1 * y2) * len1 * len2;
-
-  if(cosine > 1.0) {
-    return 0;
-  }
-  else if(cosine < -1.0) {
-    return Math.PI;
-  } else {
-    return Math.acos(cosine);
-  }
+  let x1 = a[0], y1 = a[1], x2 = b[0], y2 = b[1],
+  // mag is the product of the magnitudes of a and b
+  mag = Math.sqrt(x1*x1 + y1*y1) * Math.sqrt(x2*x2 + y2*y2),
```

Comment:
[`Math.hypot`](https://github.com/toji/gl-matrix/pull/357) could be used here.

Reply:
In my tests `Math.sqrt` was faster than `Math.hypot`, I think it might be down to `Math.hypot` taking a variable number of args.

Reply:
Thanks for testing it. 
I should probably test the other usages of `Math.hypot` again, since either the performance has changed or it's relative performance depends on something like the number of arguments.

Reply:
FF (71) no major difference, Chrome (79) showed a large gap.